src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private T[] arrayList;
+    private int size;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[10];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == arrayList.length) {
+            arrayList = addSize(arrayList);
+        }
+        arrayList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == arrayList.length) {
```

Comment:
You use this comparison logic twice. It would be better to put it in a separate method.